### ISSUE TITLE
Fast fail backup on leadership change

### ISSUE THREAD

**Issue by bradnicholson** [pgbackrest/pgbackrest#907]
If pgBackRest is running against the primary, and leadership is changed during the backup, the backup will continue until the end of the backup process then fail when trying to call pg_stop_backup against a read only database, such as:

```
INFO: full backup size = 36.3MB
INFO: execute non-exclusive pg_stop_backup() and wait for all WAL segments to archive
ASSERT: [025]: check 'PQgetResult(this->connection) == NULL' failed
        STACK TRACE:
        postgres/client:pgClientQuery:329:(debug log level required for parameters)
        perl/exec:perlExec:(debug log level required for parameters)
        main:main:(debug log level required for parameters)
INFO: http statistics: objects 1, sessions 1, requests 17, retries 0, closes 0
INFO: backup command end: aborted with exception [025]
```

It would be good if pgBackRest detected the leadership change and failed the backup when it happens.  This would be especially useful in auto-failover environments such as Patroni.

Discussed with @sfrost on Slack.

**Comment by sfrost** [comment]
Seems like something we could maybe do on the connection that's being held open where the pg_start_backup() was run, maybe during the keep-alive..?

**Comment by dwsteele** [comment]
I think checking `PQstatus()` should do the trick, but protocol keep-alive is probably not the best place since that is performed every `protocol-timeout / 2` seconds.

Once a minute maybe?  30 seconds?

**Comment by bradnicholson** [comment]
I don't think it needs to be terribly fast.  It's really to avoid a backup running for an extended time after the switchover.

**Comment by sfrost** [comment]
I like once a minute.  I'm surprised keep-alive is performed so rarely, actually..

**Comment by dwsteele** [comment]
This was fixed in bd2ba80, way back in 2.37.
